aksk/pkg/iam/driver/keystone/keystone.go

```diff
@@ -220,3 +244,56 @@ func (iam *KeystoneIam) DownloadAkSk(ctx context.Context, in *pb.GetAkSkRequest)
 	json.Unmarshal(bodyBytes, &akskListout)
 	return akskListout, nil
 }
+
+func (iam *KeystoneIam) isValidUserId(userId string, token string) bool {
+
+	// Validate userId is legitimate
+	keystoneURL := PROTOCOL + iam.Host + USER_DETAILS
+	getreq, err := http.NewRequest(GET, keystoneURL+userId, bytes.NewBuffer(nil))
+	if err != nil {
+		log.Error("Error in validating the UserId")
+		return false
+	}
+	getreq.Header.Add(AUTH_TOKEN, token)
+	getreq.Header.Set(CONTENT_TYPE, APPL_JSON)
+
+	var validationResponse *http.Response
+	validationResponse, err = iam.Client.Do(getreq)
+	if err != nil {
+		log.Error("Error in validating TenantId", err)
+		return false
+	}
+
+	if validationResponse.StatusCode == 200 {
+		log.Info("UserId is Valid")
+		return true
+	}
+
+	return false
+}
+
+func (iam *KeystoneIam) isValidTenantId(tenantId string, token string) bool {
```

Comment:
isValidTenantId or isValidUserId more or less both are same. You may remove any of these to avoid code duplicacy.

Reply:
these are different, as they are invoking different APIs on keystone.

Reply:
Both are same and just differ by userId/TenantId parameters.
So, overall only one method can solve both purpuse.

Reply:
fixed